anndata/tests/helpers.py

```diff
@@ -759,21 +758,30 @@ def shares_memory_sparse(x, y):
     ),
 ]
 
+try:
+    import zarr
 
-class AccessTrackingStore(zarr.DirectoryStore):
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-        self._access_count = {}
+    class AccessTrackingStore(zarr.DirectoryStore):
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, **kwargs)
+            self._access_count = {}
 
-    def __getitem__(self, key):
-        for tracked in self._access_count:
-            if tracked in key:
-                self._access_count[tracked] += 1
-        return super().__getitem__(key)
+        def __getitem__(self, key):
+            for tracked in self._access_count:
+                if tracked in key:
+                    self._access_count[tracked] += 1
+            return super().__getitem__(key)
 
-    def get_access_count(self, key):
-        return self._access_count[key]
+        def get_access_count(self, key):
+            return self._access_count[key]
 
-    def set_key_trackers(self, keys_to_track):
-        for k in keys_to_track:
-            self._access_count[k] = 0
+        def set_key_trackers(self, keys_to_track):
+            for k in keys_to_track:
+                self._access_count[k] = 0
+except ImportError:
+
+    class AccessTrackingStore:
+        def __init__(self) -> None:
+            raise ImportError(
+                "zarr must be imported to create an `AccessTrackingStore` instance."
+            )
```

Comment:
Not a huge fan of this, but I think the other option is to create `AccessTrackingStore` based on some dummy base class, which will error out on construction with a less helpful error.  And if we want to raise an error, we're sort of back to this.